### ISSUE TITLE
Shell escape for multiple file creation?

### ISSUE THREAD

**Issue by communistkiro** [tectonic-typesetting/tectonic#836]
My tex package [https://github.com/communistkiro/dmlb](dmlb) has an option to enable graphing of the data. When I first started using tectonic, shell-escape wasn't a thing, but now it is, and  get the following error:

```
! Package tikz Error: Sorry, the system call 'xelatex -shell-escape -halt-on-er
ror -interaction=batchmode -jobname "./graphs/88" "\def\tikzexternalrealjob{gra
phmedaddy}\input{graphmedaddy}"' did NOT result in a usable output file './grap
hs/88' (expected one of .pdf:.jpg:.jpeg:.png:.bmp:). Please verify that you hav
e enabled system calls. For pdflatex, this is 'pdflatex -shell-escape'. Sometim
es it is also named 'write 18' or something like that. Or maybe the command sim
ply failed? Error messages can be found in './graphs/88.log'. If you continue n
ow, I'll try to typeset the picture.
```

1--87 have been already created, hence the skip. I do not know if it is possible to run tectonic seperately for every file in specific way so as to get the .pdf, or if I'm missing something. Little-to-nothing of note has changed within the .tex and .sty files, I house locally; the command I'm running is `tectonic --keep-intermediates --keep-logs --print --reruns 0 -Z shell-escape graphmedaddy.tex && tectonic --keep-intermediates --keep-logs --print --reruns 0 -Z shell-escape fmsuf.tex;` 88.dat I've verified is proper and valid, as much as the previous 87 are.

I'll provide additional information if required to aid diagnosis.

**Comment by communistkiro** [comment]
Not quite, TeX _is_ n times, but I run the command once to create the graphs. Starting without any other files than the two .tex and the .sty files, one would have to: 
- run xelatex once without my [graph] option to create all the .dat files and the basic document;
- run xelatex once with shell-escape option on the graphing .tex, this created a .pdf from every .dat without one;
- run xelatex once with [graph] option, wherein .pdf's are merely included, not not created (I cannot remember why I'd split this into two files, but I wrote it awhiles back).

_If you generate that 88th graph using xelatex, does the overall document build work correctly in Tectonic?_
Deleting all .dat files after 88, the graphme.pdf including all separate graphs is generated, however, since .dat files are regenerated on compilation, building the main .tex with the [graph] option results in everything up to the 88 graph being rendered correctly based on the output of tectonic: ...yaddayadda.. `Output written on fmsuf.xdv (229 pages, 24122436 bytes).`, halting on the exact same tikz error as in the OP.

Using shell shenanigans, probably everything could be automated, but the idea behind the package was to be standalone and without fuss: enter data, run xelatex in the above order with the relevant options, gets graphs from the data. However, I don't know how to map the `xelatex -shell-escape -halt-on-error -interaction=batchmode -jobname "./graphs/88" "\def\tikzexternalrealjob{graphmedaddy}\input{graphmedaddy}"` command to tectonic, like mentioned the jobname spec is one of the things.

Regarding a MWE:
- clone my repo; `template.tex` file has values include for testing purposes, edit it to delete the _% CALENDAR_ section right after \begin{document}, remove the [graph] option, and run `tectonic --keep-intermediates --keep-logs --print --reruns 1 template.tex` with the 2 files of 20-something days plus as many in the many .tex to get 3 .dat files in graphs/: 0, 1, and 81 (each dat files spans 28 days as per the option in template.tex, I date skip to 2006-02-01 is ~ 88*28 days);
- run `tectonic --keep-intermediates --keep-logs --print --reruns 0 -Z shell-escape graphmedaddy.tex && tectonic --keep-intermediates --keep-logs --print --reruns 0 -Z shell-escape template.tex` ; I receive the same error but for 1 instead of 88.

**Comment by kpym** [comment]
This is related to #852
